docker-entrypoint.sh

```diff
@@ -267,6 +275,37 @@ docker_sql_escape_string_literal() {
 	echo "${escaped//\'/\\\'}"
 }
 
+# Creates replication user
+create_replica_user() {
+	if [ "$MARIADB_REPLICATION_MODE" = "master" ] && [ -n "$MARIADB_REPLICATION_USER" ]; then
+		if [ -n "$MARIADB_REPLICATION_PASSWORD" ] || [ -n "$MARIADB_REPLICATION_PASSWORD_HASH" ]; then
+			mysql_note "Creating user ${MARIADB_REPLICATION_USER}"
+			if [ -n  "$MARIADB_REPLICATION_PASSWORD_HASH" ]; then
+				echo "CREATE USER '$MARIADB_REPLICATION_USER'@'%' IDENTIFIED BY PASSWORD '$MARIADB_REPLICATION_PASSWORD_HASH';"
+			else
+				# SQL escape the user password, \ followed by '
+				local userPasswordEscaped
+				userPasswordEscaped=$( docker_sql_escape_string_literal "${MARIADB_REPLICATION_PASSWORD}" )
+				echo "CREATE USER '$MARIADB_REPLICATION_USER'@'%' IDENTIFIED BY '$userPasswordEscaped';"
+			fi
+			echo "GRANT REPLICATION REPLICA ON *.* TO '$MARIADB_REPLICATION_USER'@'%';"
+		else
+			mysql_error "MARIADB_REPLICATION_PASSWORD or MARIADB_REPLICATION_PASSWORD_HASH not found to create replica user"
```

Comment:
please check `mysql_error` and `mysql_note` go to stderr. Otherwise the stdout aspect of this function interface for providing SQL is flawed.

Reply:
I've created a function `validate_replica_user` which validate first then create a replica user. Please check that out :)